opendevin/core/main.py

```diff
@@ -184,11 +184,13 @@ async def on_event(event: Event):
         else config.max_budget_per_task
     )
 
-    asyncio.run(
+    final_state = asyncio.run(
         run_agent_controller(
             agent=agent,
             task_str=task_str,
             max_iterations=args.max_iterations,
             max_budget_per_task=args.max_budget_per_task,
         )
     )
+    if final_state.last_error:
+        sys.exit(1)
```

Comment:
Please don't do this. We should only use `sys.exit(1)` in production code (tests excluded) as last-resort.

Reply:
side effects of this? It just sets the current exit code of the script.